testssl.sh

```diff
@@ -18193,68 +18193,39 @@ determine_service() {
-     local overflow_cipher1='C0,86'
-     local overflow_cipher2='C0,88'
+     # overflow_cipher must be some cipher that does not appear in TLS12_CIPHER.
+     local overflow_cipher='C0,86'
+     local -i nr_ciphers
 
      # For STARTTLS protcols not being implemented yet via sockets this is a bypass otherwise it won't be usable at all (e.g. LDAP)
      # Fixme: find out whether we can't skip this in general for STARTTLS
      [[ "$STARTTLS" =~ ldap ]] && return 0
      [[ "$STARTTLS" =~ irc ]] && return 0
 
-     debugme echo -n "${FUNCNAME[0]} starting at # of ciphers (excl. 00FF): "
-     debugme echo  "$(tr ' ' '\n' <<< "$test_ciphers" | wc -l)"
      # Only with TLS 1.2 offered at the server side it is possible to hit this bug, in practise. Thus
      # we assume if TLS 1.2 is not supported, the server has no cipher size limit bug. It still may,
      # theoretically, but in a regular check with testssl.sh we won't hit this limit with lower protocols.
-     # Upon calling this function we may know already whether TLS 1.2 is supported. If not we just
-     # check for it (and add it to the known protocols to be supported).
-     # Then we send 127 ciphers, check whether they work, and increase it by one and check again. The
-     # limit bug should occur @ 128 ciphers. To be sure we test until 129 ciphers.
+     # Upon calling this function we already know whether TLS 1.2 is supported. If TLS 1.2 is supported, we
+     # send 129 ciphers (including 00FF) and check whether it works.
 
      if [[ 1 -eq $(has_server_protocol 03) ]]; then
           SERVER_SIZE_LIMIT_BUG=false
-     elif [[ 0 -eq $(has_server_protocol 03) ]]; then
-          debugme echo "Sending 127 ciphers"
-          tls_sockets 03 "${test_ciphers}, 00,FF"
-          if [[ $? -eq 0 ]]; then
-               debugme echo "Sending 128 ciphers"
-               tls_sockets 03 "${test_ciphers}, ${overflow_cipher1}, 00,FF"
-               if [[ $? -ne 0 ]]; then
-                    SERVER_SIZE_LIMIT_BUG=true
+     else
+          if [[ "$DEBUG" -ge 1 ]]; then
+               nr_ciphers="$(tr ' ' '\n' <<< "${overflow_cipher}, $TLS12_CIPHER" | sed -e '/^$/d' | wc -l)"
```

Comment:
can you please check whether this works under Darwin? IIRC FreeBSD+Darwin add a blank sometimes with ``wc -l``. ``count_lines()`` would be preferred otherwise.

Reply:
Hi Dirk,

It worked for me on MacOS 10.14.6. `$TLS12_CIPHER` includes a blank line, so I add to add `sed -e '/^$/d'` to remove any blank lines before counting. Perhaps that also prevents any problems with FreeBSD+Darwin